spec/bolt_spec/run_spec.rb

```diff
@@ -69,16 +69,70 @@
 
   describe 'run_plan' do
     it 'should run a plan' do
-      result = run_plan('sample::single_task', { 'nodes' => 'ssh' }, config: config_data, inventory: inventory_data)
+      result = run_plan('sample::single_task', { 'nodes' => 'ssh' }, options)
       expect(result['status']).to eq('success')
       data = result['value'][0]
       expect(data['status']).to eq('success')
     end
 
     it 'should return a failure' do
-      result = run_plan('error::run_fail', { 'target' => 'ssh' }, config: config_data, inventory: inventory_data)
+      result = run_plan('error::run_fail', { 'target' => 'ssh' }, options)
       expect(result['status']).to eq('failure')
       expect(result['value']['kind']).to eq('bolt/run-failure')
     end
   end
+
+  context 'with a target that has a puppet-agent installed' do
+    def run_root_config
+      { 'ssh' => root_config['ssh'] }
+    end
+
+    before(:all) do
+      result = run_task('puppet_agent::version', 'ssh', inventory: conn_inventory, config: run_root_config)
+      expect(result.first['status']).to eq('success')
+      unless result.first['result']['version']
+        result = run_task('puppet_agent::install', 'ssh', inventory: conn_inventory, config: run_root_config)
```

Comment:
I'd be comfortable running these tests on a single target rather than making sure we get puppet 5 and puppet 6.

Reply:
Maybe that's what we're doing with the `ssh` target?

Reply:
Yeah, I was originally going to test against puppet 5 and puppet 6 but went with just testing against the single `ubuntu_node` `bolt-spec` image.